dlib/CMakeLists.txt

```diff
@@ -136,10 +136,12 @@ endmacro()
 
 
 # Suppress superfluous randlib warnings about libdlib.a having no symbols on MacOSX.
-if (APPLE)
+if (CMAKE_C_COMPILER_ID STREQUAL "AppleClang")
    set(CMAKE_C_ARCHIVE_CREATE   "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
-   set(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
    set(CMAKE_C_ARCHIVE_FINISH   "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
+endif()
+if (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+   set(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
```

Comment:
Hi, why do we need to close and open the if block? Aren't the conditions the same?
(I'm on my phone, so maybe I missed something)